pkg/tbtc/node.go

```diff
@@ -1075,6 +1089,172 @@ func processCoordinationResult(node *node, result *coordinationResult) {
 	}
 }
 
+// archiveClosedWallets archives closed or terminated wallets.
+func (n *node) archiveClosedWallets() error {
+	getClosedWallets := func(walletPublicKeyHashes [][20]byte) (
+		closedWallets [][20]byte,
+	) {
+		for _, walletPublicKeyHash := range walletPublicKeyHashes {
+			walletChainData, err := n.chain.GetWallet(walletPublicKeyHash)
```

Comment:
Once we have the `walletID` cached (see another comment), I think we can call `n.chain.IsWalletRegistered` instead of inspecting the wallet state in the `Bridge`. This way we have the same condition for startup and event-based archiving.

Reply:
Done in 95c56bb097cd97d7d22d717ded4226e6a5755788.
Although I had to use `n.chain.CalculateWalletID(walletPublicKey)` instead of the stored `walletID`, as the walletID is stored in wallet cache and is not returned from `getWalletByID` or `getWalletByPublicKeyHash`.